.github/workflows/pr-lint.yml

```
@@ -0,0 +1,16 @@
+name: PR Quality Check
+on:
+  pull_request:
+    types: [opened, synchronize, reopened, edited]
+jobs:
+  check-pr-name:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check ticket name conforms to requirements
+        run: 'echo ${{ github.event.pull_request.title }} | grep -i -E -q "^(niad)-[0-9]+: [a-z]+|(dependabot\/)"'
+
+  check-branch-name:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check branch name conforms to requirements
+        run: 'echo ${{ github.event.pull_request.head.ref }} | grep -i -E -q "^(feature|bug)\/niad-[0-9]+-[a-z]+"'
```

Think you stripped out a bit too much 😁
We want to allow any positive number of _dash letters_ sequences, so we will need a group to represent that repeatable structure.
```suggestion
        run: 'echo ${{ github.event.pull_request.head.ref }} | grep -i -E -q "^(feature|bug)\/niad-[0-9]+(-[a-z]+)+"'
```

---

Made new capture group as suggested